src/plots/geo/geo2.js

```diff
@@ -0,0 +1,612 @@
+        Fx.loneUnhover(fullLayout._toppaper);
+    });
+
+    framework.on('click', function() {
+        Fx.click(gd, d3.event);
+    });
+};
+
+proto.makeFramework = function() {
+    var _this = this;
+    var fullLayout = _this.graphDiv._fullLayout;
+    var clipId = 'clip' + fullLayout._uid + _this.id;
+
+    var defGroup = fullLayout._defs.selectAll('g.clips')
+        .data([0]);
+    defGroup.enter().append('g')
+        .classed('clips', true);
+
+    _this.clipDef = defGroup.append('clipPath')
+        .attr('id', clipId);
+
+    _this.clipRect = this.clipDef.append('rect');
+
+    _this.framework = d3.select(_this.container).append('g')
+        .attr('class', 'geo ' + _this.id)
+        .call(Drawing.setClipUrl, clipId);
+
+    _this.xaxis = {
+        _id: 'x',
+        c2p: function(v) {
+            return (_this.projection(v) || [])[0] - _this.xaxis._offset;
+        }
+    };
+
+    _this.yaxis = {
+        _id: 'y',
+        c2p: function(v) {
+            return (_this.projection(v) || [])[1] - _this.yaxis._offset;
+        }
+    };
+
+    // mock axis for hover formatting
+    _this.mockAxis = {
+        type: 'linear',
+        showexponent: 'all',
+        exponentformat: 'B'
+    };
+    Axes.setConvert(_this.mockAxis, fullLayout);
+
+    var geoLayout = fullLayout[_this.id];
+    var center = geoLayout.center || {};
+    var projLayout = geoLayout.projection;
+    var rotation = projLayout.rotation || {};
+
+    if(geoLayout._isScoped) {
+        _this.viewInitial = {
+            'center.lon': center.lon,
+            'center.lat': center.lat,
+            'projection.scale': projLayout.scale
+        };
+    } else if(geoLayout._isClipped) {
+        _this.viewInitial = {
+            'projection.scale': projLayout.scale,
+            'projection.rotation.lon': rotation.lon,
+            'projection.rotation.lat': rotation.lat
+        };
+    } else {
+        _this.viewInitial = {
+            'center.lon': center.lon,
+            'center.lat': center.lat,
+            'projection.scale': projLayout.scale,
+            'projection.rotation.lon': rotation.lon
+        };
+    }
+};
+
+// [hot code path] (re)draw all paths which depend on the projection
+proto.render = function() {
+    var projection = this.projection;
+    var pathFn = projection.getPath();
+    var k;
+
+    function translatePoints(d) {
+        var lonlatPx = projection(d.lonlat);
+        return lonlatPx ?
+            'translate(' + lonlatPx[0] + ',' + lonlatPx[1] + ')' :
+             null;
+    }
+
+    function hideShowPoints(d) {
+        return projection.isLonLatOverEdges(d.lonlat) ? 'none' : null;
+    }
+
+    for(k in this.basePaths) {
+        this.basePaths[k].attr('d', pathFn);
+    }
+
+    for(k in this.dataPaths) {
+        this.dataPaths[k].attr('d', function(d) { return pathFn(d.geojson); });
+    }
+
+    for(k in this.dataPoints) {
+        this.dataPoints[k]
+            .attr('display', hideShowPoints)
```

Comment:
> marker.opacity wasn't applied properly on scattergeo bubbles
> Was it just the default being applied improperly, or was opacity not supported before?

This here is why suddenly `marker.opacity` is applied correctly in `scattergeo` traces. [Previously](https://github.com/plotly/plotly.js/blob/549ee3866d86b64bd34a574573bd5596a57c1ed0/src/plots/geo/geo.js#L447) we used `opacity` 0 :left_right_arrow:  1 to hide points that over the map's edges which erroneously overrode whatever `Drawing.pointStyle` set.  See https://github.com/plotly/plotly.js/pull/1861#discussion_r126199803 for why we now use `display` hide/show points.

> @etpinard and I discussed bubble opacity a bit more - I was on the fence but he won me over with keeping symmetry with regular scatter. So the behavior introduced here is great.

:ok_hand: